src/server/game/AI/CoreAI/ConversationAI.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TRINITY_CONVERSATIONAI_H
+#define TRINITY_CONVERSATIONAI_H
+
+#include "Define.h"
+#include "ObjectGuid.h"
+
+class Conversation;
+class Unit;
+class Player;
+
+class TC_GAME_API ConversationAI
+{
+        uint32 _scriptId;
+
+    protected:
+        Conversation* const convo;
+    public:
+        explicit ConversationAI(Conversation* c, uint32 scriptId = {});
+        virtual ~ConversationAI();
+
+        // Called when the Conversation has just been initialized, just before added to map
+        virtual void OnInitialize() {}
```

Comment:
instead of using empty functions, we can mark them in the base class with ```cpp = default``` where desired, such as OnInitialize, OnCreate, and OnStart.

Reply:
What? No, only "special members" such as contructors/destructor/assignment operator/operator=/operator<=> can be defaulted

Reply:
as far as I know, virtual functions can be set as default if no behavior is required, unless overriding is expected

Reply:
https://en.cppreference.com/w/cpp/language/member_functions#Special_member_functions